src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -142,6 +143,53 @@ private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, P
             }
         }
 
+        private void ParseServerVariables(XElement serverVariables, UrlRewriteRuleBuilder builder)
+        {
+            if (serverVariables == null)
+            {
+                return;
+            }
+
+            foreach (var serverVariable in serverVariables.Elements(RewriteTags.Set))
+            {
+                ParseServerVariable(serverVariable, builder);
+            }
+        }
+
+        private void ParseServerVariable(XElement serverVariable, UrlRewriteRuleBuilder builder)
+        {
+            try
+            {
+                string prefix;
+                ServerVariableType type;
+                var name = serverVariable.Attribute(RewriteTags.Name).Value;
+
+                if (name.StartsWith(ServerVariableConstants.RequestHeaderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    prefix = ServerVariableConstants.RequestHeaderPrefix;
+                    type = ServerVariableType.Request;
+                }
+                else if (name.StartsWith(ServerVariableConstants.ResponseHeaderPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
IIS/ARR definitely allow setting a response header as a part of an inbound rule. i emulated the behavior with the configuration we have in `applicationHost.config`. i don't have any outbound rules in my config otherwise i would've implemented that feature as well. ;)